File: 30DaysOfCode/day15_linkedLists/linkedLists.php
<?php
class Node {
  public $data;
  public $next;

  function __construct($d) {
    $this->data = $d;
    $this->next = NULL;
  }
}

class Solution {
  function insert($head, $data) {
    $newNode = new Node($data);
        
    if ($head == NULL) {
      return $newNode;
    } else {
      $node = $head;
            
      while($node->next != NULL) {
        $node = $node->next;
      }
      $node->next = $newNode;
            
      return $head;
    }
  }
  
  function display($head) {
    $start = $head;
        
    while($start) {
      echo $start->data, ' ';
      $start = $start->next;
    }
  }
}

$T = intval(fgets(STDIN));
$head = null;
$mylist = new Solution();

while($T-- > 0) {
  $data = intval(fgets(STDIN));
  $head = $mylist->insert($head, $data);
}

$mylist->display($head);
?>

File: 30DaysOfCode/day15_linkedLists/linkedLists.rb
class Node
    attr_accessor :data, :next

    def initialize data
        @data = data
        @next = nil
    end
end

class Solution
    def insert(head, value)
        newNode = Node.new(value)
        
        if (head == nil)
            return newNode
        else
            node = head
            
            while (node.next) do
                node = node.next 
            end
            node.next = newNode
            
            return head
        end
    end
    
    def display(head)
       current = head
       while current 
          print current.data, " "
          current = current.next
       end
   end
end

       
mylist = Solution.new
head = nil
T = gets.to_i

for i in 1..T
    data = gets.to_i
    head = mylist.insert(head, data)
end

mylist.display(head)
